reports-scheduler/src/test/kotlin/com/amazon/opendistroforelasticsearch/reportsscheduler/rest/ReportDefinitionIT.kt

```diff
@@ -0,0 +1,303 @@
+                        "title":"title",
+                        "textDescription":"textDescription",
+                        "htmlDescription":"optional htmlDescription",
+                        "channelIds":["optional_channelIds"]
+                    }
+                }
+            }
+            """.trimIndent()
+    }
+
+    fun `test create, get, update, delete report definition`() {
+        var reportDefinitionRequest = constructReportDefinitionRequest()
+        var reportDefinitionResponse = executeRequest(
+            RestRequest.Method.POST.name,
+            "$BASE_REPORTS_URI/definition",
+            reportDefinitionRequest,
+            RestStatus.OK.status
+        )
+        var reportDefinitionId = reportDefinitionResponse.get("reportDefinitionId").asString
+        Assert.assertNotNull("reportDefinitionId should be generated", reportDefinitionId)
+        Thread.sleep(100)
+
+        reportDefinitionRequest = constructReportDefinitionRequest(
+            """
+                "trigger":{
+                    "triggerType":"Download"
+                },
+            """.trimIndent()
+        )
+        reportDefinitionResponse = executeRequest(
+            RestRequest.Method.POST.name,
+            "$BASE_REPORTS_URI/definition",
+            reportDefinitionRequest,
+            RestStatus.OK.status
+        )
+        reportDefinitionId = reportDefinitionResponse.get("reportDefinitionId").asString
+        Assert.assertNotNull("reportDefinitionId should be generated", reportDefinitionId)
+        Thread.sleep(3000)
+
+        val reportDefinitionsResponse = executeRequest(
+            RestRequest.Method.GET.name,
+            "$BASE_REPORTS_URI/definitions",
+            "",
+            RestStatus.OK.status
+        )
+        Assert.assertEquals(2, reportDefinitionsResponse.get("totalHits").asInt)
+        Thread.sleep(100)
+
+        val newName = "updated_report"
+        reportDefinitionRequest = constructReportDefinitionRequest(name = newName)
+        reportDefinitionResponse = executeRequest(
+            RestRequest.Method.PUT.name,
+            "$BASE_REPORTS_URI/definition/$reportDefinitionId",
+            reportDefinitionRequest,
+            RestStatus.OK.status
+        )
+        Assert.assertEquals(reportDefinitionId, reportDefinitionResponse.get("reportDefinitionId").asString)
+        Thread.sleep(100)
+
+        reportDefinitionResponse = executeRequest(
+            RestRequest.Method.GET.name,
+            "$BASE_REPORTS_URI/definition/$reportDefinitionId",
+            "",
+            RestStatus.OK.status
+        )
+        Assert.assertEquals(
+            reportDefinitionId,
+            reportDefinitionResponse.get("reportDefinitionDetails").asJsonObject.get("id").asString
+        )
+        Assert.assertEquals(
+            newName,
+            reportDefinitionResponse.get("reportDefinitionDetails").asJsonObject.get("reportDefinition").asJsonObject.get(
```

Comment:
may be indent with second .get() in new line?

Reply:
updated